File: common-ui/src/commonMain/kotlin/com/carlom/klardrop/theme/KdRadii.kt
package com.carlom.klardrop.theme

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

// ---------------------------------------------------------------------------
// KdRadii — corner radius tokens + corresponding shapes from spec/tokens.css
// ---------------------------------------------------------------------------

data class KdRadii(
    /** 6 dp — status dot outline, chip tints */
    val xs: Dp = 6.dp,
    /** 10 dp — file icon tile inside a file card */
    val sm: Dp = 10.dp,
    /** 14 dp — buttons, banners, selected state */
    val md: Dp = 14.dp,
    /** 18 dp — device row, chat bubble, file card */
    val lg: Dp = 18.dp,
    /** 24 dp — bottom sheet, dialog, hero card */
    val xl: Dp = 24.dp,

    /** Fully-rounded: visibility pill, message input, send button */
    val shapeXs: RoundedCornerShape = RoundedCornerShape(6.dp),
    val shapeSm: RoundedCornerShape = RoundedCornerShape(10.dp),
    val shapeMd: RoundedCornerShape = RoundedCornerShape(14.dp),
    val shapeLg: RoundedCornerShape = RoundedCornerShape(18.dp),
    val shapeXl: RoundedCornerShape = RoundedCornerShape(24.dp),
    val shapePill: RoundedCornerShape = RoundedCornerShape(50),
)

val KdDefaultRadii = KdRadii()

File: iosApp/iosApp/Design/KdRadii.swift
import SwiftUI

// ---------------------------------------------------------------------------
// KdRadii — corner radius tokens (SwiftUI translation of compose-ui KdRadii.kt)
// iOS 14.1+ compatible.
// ---------------------------------------------------------------------------

enum KdRadii {
    /// 6 pt — status dot outline, chip tints
    static let xs: CGFloat = 6
    /// 10 pt — file icon tile inside a file card
    static let sm: CGFloat = 10
    /// 14 pt — buttons, banners, selected state
    static let md: CGFloat = 14
    /// 18 pt — device row, chat bubble, file card
    static let lg: CGFloat = 18
    /// 24 pt — dialog, hero card
    static let xl: CGFloat = 24
    /// 28 pt — bottom sheet top corner
    static let sheet: CGFloat = 28
}

// MARK: - Convenience shapes

enum KdShape {
    static let xs = RoundedRectangle(cornerRadius: KdRadii.xs, style: .continuous)
    static let sm = RoundedRectangle(cornerRadius: KdRadii.sm, style: .continuous)
    static let md = RoundedRectangle(cornerRadius: KdRadii.md, style: .continuous)
    static let lg = RoundedRectangle(cornerRadius: KdRadii.lg, style: .continuous)
    static let xl = RoundedRectangle(cornerRadius: KdRadii.xl, style: .continuous)
    /// Sheet: only top corners rounded (iOS 14 compatible — uses Path-based shape)
    static let sheet = TopRoundedRectangle(radius: KdRadii.sheet)
    /// Pill / fully-rounded capsule
    static let pill = Capsule()
}

// MARK: - TopRoundedRectangle (iOS 14+ compatible, top corners only)

struct TopRoundedRectangle: Shape {
    let radius: CGFloat
    func path(in rect: CGRect) -> Path {
        var path = Path()
        path.move(to: CGPoint(x: rect.minX + radius, y: rect.minY))
        path.addLine(to: CGPoint(x: rect.maxX - radius, y: rect.minY))
        path.addQuadCurve(
            to: CGPoint(x: rect.maxX, y: rect.minY + radius),
            control: CGPoint(x: rect.maxX, y: rect.minY)
        )
        path.addLine(to: CGPoint(x: rect.maxX, y: rect.maxY))
        path.addLine(to: CGPoint(x: rect.minX, y: rect.maxY))
        path.addLine(to: CGPoint(x: rect.minX, y: rect.minY + radius))
        path.addQuadCurve(
            to: CGPoint(x: rect.minX + radius, y: rect.minY),
            control: CGPoint(x: rect.minX, y: rect.minY)
        )
        path.closeSubpath()
        return path
    }
}
